### ISSUE TITLE
Public code blocked where it shouldn't

### ISSUE THREAD

**Issue by svenbledt** [microsoft/vscode-copilot-release#532]
- VS Code Version: latest
- OS Version: ![image](https://github.com/microsoft/vscode-copilot-release/assets/8340034/6c394c8f-43b1-4a91-aedb-7d0745e26198)
- Logs: 

Steps to Reproduce:

1. select a code that is obviously from a public repo
2. translate it to english or whatever language it is not in
3. tell github copilot to translate it to the language the public file already has.

![image](https://github.com/microsoft/vscode-copilot-release/assets/8340034/59d1605c-e321-4932-b4bc-a20ac1bf1d46)

My Settings:

![image](https://github.com/microsoft/vscode-copilot-release/assets/8340034/32124120-21fd-428a-aa7d-b52b4d2c84e8)

**Comment by dbdean** [comment]
I'm getting the public code blocked response on every single question at the moment. I do have it set to allow, but I don't think the problem is it is only ignoring my settings there. I think something has changed that it thinks every single response has public code, and it ignores the users setting as well.

**Comment by NullpoGah** [comment]
Same goes for me, every response is filtered for some reason

**Comment by ismir2020** [comment]
Same here. Wanted to give me new code for my login.blade.php with the same design of register.blade.php then the answer "Sorry, the response matched public code". I tried already to block matching public code at the github settings but still facing this issue.

The code isnt anywhere on a repo on github or somewhere else. The reset password could be in some cases with the same layout cause its needed for nearly every application....

The answer here (https://github.com/microsoft/vscode-copilot-release/issues/295#issuecomment-1760046462) doesnt make any changes...

**Comment by lramos15** [comment]
Can you please provide the logs from `GitHub Copilot Chat` output channel. This will help us investigate what is going on in the backend with these requests

**Comment by digitarald** [comment]
Looks like off-topic detection incorrectly marks responses as filtered, which leads to the wrong error.

We are working on a fix. Meanwhile, public-code error responses mean that the off-topic detection kicked in and you can solve this by rephrasing your question.

**Comment by sharonlo** [comment]
Fix ix happening here: https://github.com/github/copilot-api/issues/1817

**Comment by isi2010** [comment]
How to get the output logs of it?

rephrasing doesn’t make any difference. I tried several times with different prompts and everytime it starts creating the answer and then stopping with the error message.

And if set the setting to „block“ then it don’t gonna generate codes anymore or it only doesn’t check public code ?

**Comment by lramos15** [comment]
> How to get the output logs of it?
> 
> rephrasing doesn’t make any difference. I tried several times with different prompts and everytime it starts creating the answer and then stopping with the error message.
> 
> And if set the setting to „block“ then it don’t gonna generate codes anymore or it only doesn’t check public code ?

`CTRL/CMD + SHIFT + U` then go to the dropdown and select "GitHub Copilot Chat" and copy the info

**Comment by ismir2020** [comment]
> > How to get the output logs of it?
> > rephrasing doesn’t make any difference. I tried several times with different prompts and everytime it starts creating the answer and then stopping with the error message.
> > And if set the setting to „block“ then it don’t gonna generate codes anymore or it only doesn’t check public code ?
> 
> `CTRL/CMD + SHIFT + U` then go to the dropdown and select "GitHub Copilot Chat" and copy the info

My console output:

`2023-11-03T19:49:01.667Z [INFO] [streamMessages] message 0 returned. finish reason: [stop]
2023-11-03T19:49:01.668Z [INFO] [streamChoices] request done: headerRequestId: [0066d12e-e5ab-4152-a2b9-b3401b1043b4] model deployment ID: [x2191bfb807ec]
2023-11-03T19:49:44.464Z [INFO] [chat fetch] engine https://api.githubcopilot.com/chat
2023-11-03T19:49:44.464Z [INFO] [chat fetch] modelMaxTokenWindow 4096
2023-11-03T19:49:44.464Z [INFO] [chat fetch] chat model gpt-4
2023-11-03T19:49:46.518Z [INFO] [chat fetch] request.response: [https://api.githubcopilot.com/chat/completions], took 2053 ms
2023-11-03T19:49:46.519Z [INFO] [streamMessages] message 0 returned. finish reason: [content_filter]
2023-11-03T19:49:46.520Z [INFO] [streamChoices] request done: headerRequestId: [ddc1a7c7-da96-49a1-ac45-9a5dedf6f80d] model deployment ID: [x3b0892d9e5fc]`

Prompt (originaly in german): I want to create an interactive table for leads where the user can move columns, add new columns, sort the table by columns, perform a live search, and have lazy loading for large amounts of data. Please create the necessary steps for me step by step and wait after each file for my prompt "continue" to create the next file. When you are done, say "Finished".

**Comment by lramos15** [comment]
@ismir2020 You have the public code match setting disabled in your GH settings or via the organization that granted you copilot access?

**Comment by isi2010** [comment]
Ive Set it to allow and to block. Tested with both settings value and both getting this error. Don’t have Organisation profile. Followed the documentation for allowing or blocking this setting under Settings -> Copilot -> allow/block

**Comment by lramos15** [comment]
> Ive Set it to allow and to block. Tested with both settings value and both getting this error. Don’t have Organisation profile. Followed the documentation for allowing or blocking this setting under Settings -> Copilot -> allow/block

Do you have some logs when you have it set to Allow that I can check?

**Comment by isi2010** [comment]
> > Ive Set it to allow and to block. Tested with both settings value and both getting this error. Don’t have Organisation profile. Followed the documentation for allowing or blocking this setting under Settings -> Copilot -> allow/block
> 
> Do you have some logs when you have it set to Allow that I can check?

No cause I cleared the chat cause I thought it didn’t get the settings update from allow to block. But I can try to run the commands again with Allow tomorrow and send you the console output

**Comment by akhilesh84** [comment]
Facing same issue since yesterday evening.
2023-11-05T11:36:56.957Z [INFO] [chat] copilot token chat_enabled: true
2023-11-05T11:36:56.957Z [INFO] [auth] Got Copilot token for akhilesh84
2023-11-05T11:37:13.229Z [INFO] [chat fetch] engine https://api.githubcopilot.com/chat
2023-11-05T11:37:13.229Z [INFO] [chat fetch] modelMaxTokenWindow 4096
2023-11-05T11:37:13.229Z [INFO] [chat fetch] chat model gpt-4
2023-11-05T11:37:14.163Z [INFO] [chat fetch] request.response: [https://api.githubcopilot.com/chat/completions], took 933 ms
2023-11-05T11:37:14.165Z [INFO] [chat fetch] Request ID for failed request: ,8712b723-631d-4d6b-b77b-dcc0f87642d4
2023-11-05T11:37:14.169Z [ERROR] [chat fetch] Unhandled status from server:,400,Bad Request

**Comment by NidusUmbra** [comment]
Same here, and it has been this way for a long time. I assumed I had turned public code > off for some reason without remembering why, and assumed the reason was valid, whatever it was.

However this was not the case and the setting had been to "allow public code" all along, so it is a long running issue.

I also set it to allow to use my snippets to improve the product, if that matters. I initially suspected the issue was because it had just decided to use my code in said product improvement, but then ironically immediately disallowed me from viewing my, now public, code. However since both settings were to allow, this can't have been the case. Perhaps there is a twilight zone where the snippet gets submitted but the user setting to allow use isn't verified yet?

**Comment by ryanjsalva** [comment]
As of 10:30 AM on Sunday (Nov 5), I'm receiving 404s for matching public code found in code completion.

![image_720](https://github.com/microsoft/vscode-copilot-release/assets/5559646/a9438719-e3a5-4ea8-85d7-26745b99be57)
![image_720](https://github.com/microsoft/vscode-copilot-release/assets/5559646/c5a6a275-ee7e-457d-903b-b61b20946252)

**Comment by lramos15** [comment]
@akhilesh84 I think you're getting off topic filtered (the model doesnt think your question is about code) and we're just reporting it incorrectly. Can you update to the latest pre-release and vs code insiders and give it a shot. I've fixed this bug.


@ryanjsalva That's for completions and not chat, which is not something this repo tracks. Those are tracked in GitHub's own discussion https://github.com/orgs/community/discussions/categories/copilot

**Comment by mtaran** [comment]
@svenbledt: snippy blocking was (correctly) not enabled for your request. It was instead blocked by RAI's "violence" filter, since (as you can see at the end of your screenshot), the completion was talking about guns/"waffe"="weapon".

@dbdean: I couldn't find any blocked chat responses for your account in the past week. Were you signed in as this GH user? If you can still repro the blocking, could you please include logs per above?

@ismir2020: Indeed sometimes you get blocked by the text copyright filter; it seems like this happened when the model was generating boilerplate html with included scripts (ex APIM ID 095ff90f-8815-45cb-a3f6-93d124c202d1). At other times, your chat responses seem to have been over-blocked by the RAI "sexual" filter (ex APIM ID 2d80df99-5f58-426a-adaf-cba4c9ada41c). Will try to follow up with RAI folks to report the false positives.

@NidusUmbra: Your code was over-blocked by the RAI "self-harm" filter (ex APIM ID 0bf83e34-3549-4370-bc73-ea5041dde7c9). Ditto re: reporting the false positives.

**Comment by isi2010** [comment]
> 
> @ismir2020: Indeed sometimes you get blocked by the text copyright filter; it seems like this happened when the model was generating boilerplate html with included scripts (ex APIM ID 095ff90f-8815-45cb-a3f6-93d124c202d1). At other times, your chat responses seem to have been over-blocked by the RAI "sexual" filter (ex APIM ID 2d80df99-5f58-426a-adaf-cba4c9ada41c). Will try to follow up with RAI folks to report the false positives.
> 

Okay. Wondering how this filter can apply at a registration and a login form. Maybe cause of the CDN links for js,css and so on?

How I can workaround with this ? Any suggestions?

**Comment by mtaran** [comment]
> Maybe cause of the CDN links for js,css and so on?
yes, that's exactly where it triggered

> How I can workaround with this ? Any suggestions?
you can try asking it to skip generating the CDN links and to only generate code instead

**Comment by ismir2020** [comment]
But this cdn shouldnt be filtered through any filter cause its normal coding style to use cdn links for the files. isnt it?

**Comment by svenbledt** [comment]
Same thing in latest insiders version.

with  allowed on my github profile

**Comment by akhilesh84** [comment]
> @akhilesh84 I think you're getting off topic filtered (the model doesnt think your question is about code) and we're just reporting it incorrectly. Can you update to the latest pre-release and vs code insiders and give it a shot. I've fixed this bug.
> 

its working on insiders . Thanks. 
Any timelines for fix to be made available on regular build. i updated to 1.84.1 but the issue persist

**Comment by lramos15** [comment]
> Any timelines for fix to be made available on regular build. i updated to 1.84.1 but the issue persist

They're released monthly. So start of December

**Comment by craftzneko** [comment]
Running current insiders just getting this, cant see how this matches public code

![image](https://github.com/microsoft/vscode-copilot-release/assets/662108/4df5d919-cfef-45db-bfef-03f66072be50)

**Comment by segevsh** [comment]
Getting error 400 for the past hour, tried both VSCode and VSCode insider: 


2023-11-21T20:33:36.392Z [INFO] [chat fetch] engine https://api.githubcopilot.com/chat
2023-11-21T20:33:36.392Z [INFO] [chat fetch] modelMaxTokenWindow 4096
2023-11-21T20:33:36.392Z [INFO] [chat fetch] chat model gpt-4
2023-11-21T20:33:36.558Z [INFO] [chat fetch] request.response: [https://api.githubcopilot.com/chat/completions], took 166 ms
2023-11-21T20:33:36.559Z [INFO] [chat fetch] Request ID for failed request: ,262e9f1f-c2cd-4f38-af44-bfe48074f1cc
2023-11-21T20:33:36.559Z [ERROR] [chat fetch] Unhandled status from server:,400,Bad Request


The diagnostics: 

## Copilot

- Version: 1.138.0
- Build: prod
- Editor: vscode/1.85.0-insider

## Environment

- http_proxy: n/a
- https_proxy: n/a
- no_proxy: n/a
- SSL_CERT_FILE: n/a
- SSL_CERT_DIR: n/a
- OPENSSL_CONF: n/a

## Feature Flags

- Custom Certificates: disabled
- Send Restricted Telemetry: enabled
- Chat: enabled

## Node setup

- Number of root certificates: 140
- Operating system: Darwin
- Operating system version: 22.6.0
- Operating system architecture: arm64
- NODE_OPTIONS: n/a
- NODE_EXTRA_CA_CERTS: n/a
- NODE_TLS_REJECT_UNAUTHORIZED: n/a
- tls default min version: TLSv1.2
- tls default max version: TLSv1.3

## Network Configuration

- Proxy host: n/a
- Proxy port: n/a
- Proxy auth: n/a
- Kerberos SPN: n/a
- Reject unauthorized: disabled

## Reachability

- github.com: HTTP 200 - 
- copilot-proxy.githubusercontent.com: HTTP 200 - 
- default.exp-tas.com: HTTP 200 - 

## VS Code Configuration

- HTTP proxy: 
- HTTP proxy autentication: n/a
- Proxy Strict SSL: true
- Extension HTTP proxy support: override

## Extensions

- Is `win-ca` installed?: false
- Is `mac-ca` installed?: false

**Comment by Karam-Soussi** [comment]
Same thing, the issue is present for all questions and all codes no matter if private, public, or even my own.

![image](https://github.com/microsoft/vscode-copilot-release/assets/97461047/33e9d686-e349-49f0-a2eb-6797460f771e)

**Comment by K8sKween** [comment]
This might be duplicate of #583 or related.

**Comment by dobbels** [comment]
I am still seeing a similar issue. My settings allow response to match public code and still I am seeing this as a response on many requests.

**Comment by lramos15** [comment]
Consolidating as duplicate of https://github.com/microsoft/vscode-copilot-release/issues/583

**Comment by navidrigi** [comment]
2023-12-27 11:46:55.188 [info] [chat fetch] engine https://api.githubcopilot.com/chat
2023-12-27 11:46:55.191 [info] [chat fetch] modelMaxTokenWindow 4096
2023-12-27 11:46:55.191 [info] [chat fetch] chat model gpt-4
2023-12-27 11:46:59.159 [info] [chat fetch] request.response: [https://api.githubcopilot.com/chat/completions], took 3968 ms
2023-12-27 11:46:59.160 [info] [streamMessages] message 0 returned. finish reason: [content_filter]
2023-12-27 11:46:59.161 [info] [streamChoices] request done: requestId: [927fdfc4-4fae-49f7-ae10-f64ee7ad437a] responseId: [35722e39-4d67-498a-a86d-ce4ed0a2641d] model deployment ID: [x8d3c95500d67]

**Comment by rauven** [comment]
More annoying is the fact that this affect the "autocomplete" and you don't know why (its because of this bug).

I use copilot mostly for the repetitive tasks like filling languages or create human readable text from error codes etc,

I had this list of error codes:
  ```
...
  // _structureParser.missingRequiredKey
  // accountSet.invalidAccountName
  // accountSet.invalidApiKey
  // accountSetIfArgs.missingArgs
  // accountIsSet.notSet'
...
```
And i was using copilot autofill to create an array based on that commented list with human readable error messages but it refused to fill 'accountSet.invalidApiKey' no matter what, and worse, the autocomplete doesn't even give us a public code warning like the chat.

```
...
        '_structureParser.invalidTypeNumber' => [
          'en' => 'The value provided is not a number.',
          'pt' => 'O valor fornecido não é um número.'
        ],
        'accountSet.invalidAccountName' => [
          'en' => 'The account name provided is not valid.',
          'pt' => 'O nome de conta fornecido não é válido.'
        ],
        'accountSet.invalidApiKey' => [
          
        ],
        'accountSetIfArgs.missingArgs' => [
          'en' => 'The function requires arguments.',
          'pt' => 'A função requer argumentos.'
        ],
        'accountIsSet.notSet' => [
          'en' => 'The account name and API Key are not set.',
          'pt' => 'O nome de conta e a API Key não estão definidos.'
        ],
...
```

Even if you do not enable public code (i have it and it still won't work), a lot of code will always look similar and be a coincidence that looks like public code, but shouldn't... "invalidApiKey" shouldn't be considered public code, it way too generic.
